node/src/reactor/validator.rs

```diff
@@ -334,15 +334,18 @@ impl reactor::Reactor for Reactor {
         let effect_builder = EffectBuilder::new(event_queue);
         let (net, net_effects) = SmallNetwork::new(event_queue, config.network, true)?;
 
-        let address_gossiper = Gossiper::new_for_complete_items(config.gossip);
+        let address_gossiper =
+            Gossiper::new_for_complete_items("address_gossiper", config.gossip, registry)?;
 
         let api_server = ApiServer::new(config.http_server, effect_builder);
         let deploy_acceptor = DeployAcceptor::new();
         let deploy_fetcher = Fetcher::new(config.gossip);
         let deploy_gossiper = Gossiper::new_for_partial_items(
+            "deploy_gossiper",
             config.gossip,
             gossiper::get_deploy_from_storage::<Deploy, Event>,
-        );
+            registry,
+        )?;
```

Comment:
Maybe there should be methods like `Gossiper::deploy_gossiper(config.gossip, registry)` and `Gossiper::address_gossiper(config.gossip, registry)`.

Reply:
We actually need to disambiguate the metrics with a prefix depending on the gossiper, otherwise address gossiper and deploy gossiper metrics will overwrite each other (actually the app will probably just panic due to a duplicate name being registered).

So how would you generate the unique name required instead?

Reply:
I only meant to hide it behind a function, like this:
```rust
impl Gossiper {
  fn deploy_gossiper(config, registry) -> Resul<ERror, Self> {
    Gossiper::new_for_complete_items("address_gossiper", config.gossip, registry)?
  }
```

Reply:
I would see that as fairly unconventional personally - I'd rather leave as is.

Reply:
It's just hiding some of the unnecessary complexity but fine. I won't insist.